kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -0,0 +1,208 @@
+        TestUtils.waitFor("", 1000, timeoutMilliseconds,
+                () -> {
+                    var log = kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).getLog();
+                    return log.contains(" - " + (numOfMessages - 1));
+                });
+        return kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).getLog();
+    }
+
+    private static String getPodNameByLabel(String deployNamespace, String labelKey, String labelValue, long timeoutMilliseconds) {
+        TestUtils.waitFor("Waiting for Pod with label " + labelKey + "=" + labelValue + " to appear", 1000, timeoutMilliseconds,
+                () -> {
+                    var podList = kubeClient().getClient().pods().inNamespace(deployNamespace).withLabel(labelKey, labelValue);
+                    return !podList.list().getItems().isEmpty();
+                });
+        var pods = kubeClient().getClient().pods().inNamespace(deployNamespace).withLabel(labelKey, labelValue);
+        return pods.list().getItems().get(pods.list().getItems().size() - 1).getMetadata().getName();
+    }
+
+    /**
+     * Produce message.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param topicName the topic name
+     * @param message the message
+     * @param bootstrap the bootstrap
+     */
+    public static void ProduceMessage(String deployNamespace, String topicName, String message, String bootstrap) {
+        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
+                .withImage(Constants.STRIMZI_KAFKA_IMAGE)
+                .withName("java-kafka-producer")
+                .withRestartPolicy("Never")
+                .withCommand("/bin/sh")
+                .withArgs("-c", "echo '" + message + "'| bin/kafka-console-producer.sh --bootstrap-server " + bootstrap + " --topic " + topicName)
+                .done();
+    }
+
+    /**
+     * Produce message with test clients.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param numOfMessages the num of messages
+     * @return the string
+     * @throws IOException the io exception
+     */
+    public static String produceMessageWithTestClients(String deployNamespace, String topicName, String bootstrap, String message, int numOfMessages) throws IOException {
+        LOGGER.debug("Producing {} messages in '{}' topic", numOfMessages, topicName);
+        File file = replaceStringInResourceFile("kafka-producer-template.yaml", "kafka-producer.yaml", Map.of(
+                "%BOOTSTRAP_SERVERS%", bootstrap,
+                "%TOPIC_NAME%", topicName,
+                "%MESSAGE_COUNT%", "\"" + numOfMessages + "\"",
+                "%MESSAGE%", message));
+        kubeClient().getClient().load(new FileInputStream(file)).inNamespace(deployNamespace).create();
```

Comment:
use try-with-resources and rethrow the IOException as an UncheckedIoException to avoid the signature polution

Reply:
done